From 75843ac59de341bbf727c15a9378274f3b77fab2 Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Fri, 8 Jul 2016 00:15:49 -0500
Subject: [PATCH 09/21] Starting IGA2 vertical blank early by 1 count to avoid
 odd resolution

Subtracting 1 count from CrtcVlankStart for IGA2 appears to
suppress some monitors from reporting vertical resolution
that is 1 more than the desired vertical resolution.
(i.e., 1201 rather than 1200)

Signed-off-by: Eric Kudzin <knoppix1337@yahoo.com>
Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
---
 src/via_display.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/via_display.c b/src/via_display.c
index b49ed54..939e205 100644
--- a/src/via_display.c
+++ b/src/via_display.c
@@ -2290,10 +2290,13 @@ viaIGA2SetDisplayRegister(ScrnInfoPtr pScrn, DisplayModePtr mode)
 
 
     /* Set IGA2 vertical blank start. */
-    /* Vertical Blanking Start: 2048 - 1 (max) */
+    /* Subtracting 1 from CrtcVBlankStart appears to suppress some
+     * monitors from reporting vertical resolution that is 1 more
+     * than the desired vertical resolution. (i.e., 1201 rather
+     * than 1200) */
     DEBUG(xf86DrvMsg(pScrn->scrnIndex, X_INFO,
                         "IGA2 CrtcVBlankStart: %d\n", mode->CrtcVBlankStart));
-    temp = mode->CrtcVBlankStart;
+    temp = mode->CrtcVBlankStart - 1;
 
     /* 3X5.5A[7:0] - Vertical Blanking Start Bits [7:0] */
     hwp->writeCrtc(hwp, 0x5A, temp & 0xFF);
-- 
1.9.1

